src/foundry/client-esm/nue/tour.d.mts

```diff
@@ -1,9 +1,276 @@
+  targetElement: HTMLElement | undefined | null;
+
+  /**
+   * The HTMLElement that fades out the rest of the screen
+   */
+  fadeElement: HTMLElement | undefined;
+
+  /**
+   * The HTMLElement that blocks input while a Tour is active
+   */
+  overlayElement: HTMLElement | undefined;
+
+  /**
+   * Padding around a Highlighted Element
+   * @defaultValue `10`
+   */
+  static HIGHLIGHT_PADDING: number;
+
+  /**
+   * The unique identifier of the tour
+   */
+  get id(): string | undefined;
+
+  set id(value: string | undefined);
+
+  /**
+   * The human-readable title for the tour.
+   */
+  get title(): string;
+
+  /**
+   * The human-readable description of the tour.
+   */
+  get description(): string | undefined;
+
+  /**
+   * The package namespace for the tour.
+   */
+  get namespace(): string | undefined;
+
+  set namespace(value: string | undefined);
+
+  /**
+   * The key the Tour is stored under in game.tours, of the form `${namespace}.${id}`
+   */
+  get key(): string;
+
+  /**
+   * The configuration of tour steps
+   */
+  get steps(): Tour.Step[];
+
+  /**
+   * Return the current Step, or null if the tour has not yet started.
+   */
+  get currentStep(): Tour.Step | null;
+
+  /**
+   * The index of the current step; -1 if the tour has not yet started, or null if the tour is finished.
+   */
+  get stepIndex(): number | null;
+
+  /**
+   * Returns True if there is a next TourStep
+   */
+  get hasNext(): boolean;
+
+  /**
+   * Returns True if there is a previous TourStep
+   */
+  get hasPrevious(): boolean;
+
+  /**
+   * Return whether this Tour is currently eligible to be started?
+   * This is useful for tours which can only be used in certain circumnstances, like if the canvas is active.
+   */
+  get canStart(): boolean;
+
+  /**
+   * The current status of the Tour
+   */
+  get status(): Tour.STATUS;
+
+  /**
+   * Advance the tour to a completed state.
+   */
+  complete(): Promise<void>;
+
+  /**
+   * Exit the tour at the current step.
+   */
+  exit(): void;
+
+  /**
+   * Reset the Tour to an un-started state.
+   */
+  reset(): Promise<void>;
+
+  /**
+   * Start the Tour at its current step, or at the beginning if the tour has not yet started.
+   */
+  start(): Promise<void>;
+
+  /**
+   * Progress the Tour to the next step.
+   */
+  next(): Promise<void>;
+
+  /**
+   * Reward the Tour to the previous step.
+   */
+  previous(): Promise<void>;
+
+  /**
+   * Progresses to a given Step
+   * @param stepIndex - The step to progress to
+   */
+  progress(stepIndex: number): Promise<void>;
+
+  /**
+   * Query the DOM for the target element using the provided selector
+   * @param selector - A CSS selector
+   * @returns The target element, or null if not found
+   */
+  protected _getTargetElement(selector: string): Element | null;
+
+  /**
+   * Creates and returns a Tour by loading a JSON file
+   * @param filepath The path to the JSON file
+   */
+  static fromJSON(filepath: string): Promise<Tour>;
```

Comment:
It's become customary to write ``* @remarks Returns `new this()` so needs an override per subclass.`` for a situation like this.